Reworks skittish quirk to be automatic

coiax opened the pull request:
:cl: coiax
tweak: The Skittish quirk will now cause you to automatically dive into
a locker/crate if you bump into it (although closets must be closed).
Walk to avoid this behaviour.
/:cl:

This makes the quirk more useful, while also making it more thematic,
since the "diving into" behaviour can't be disabled, only supressed by
walking.

The cost is unchanged, as the quirk in its current form is overcosted at
2 points.

The emergent effect of skittish people diving into closets when caught
into explosions is definitely a feature, and not a bug.

---

~~There is some performance work that can be done, (moving the bumping behaviour~~
~~from the closet to the mob), but only if this is popular.~~

The "skittish" code is now an element that is added/removed from /mob/livings when they
gain/lose the TRAIT_SKITTISH trait.

It handles means that instead of listening for all atoms that bump into closets, we instead
only check when a skittish person bumps into a thing.

coiax committed:
Reworks skittish quirk to be automatic

:cl: coiax
tweak: The Skittish quirk will now cause you to automatically dive into
a locker/crate if you move into it while it is closed. Walk to avoid
this behaviour.
/:cl:

This makes the quirk more useful, while also making it more thematic,
since the "diving into" behaviour can't be disabled, only supressed by
walking.

The cost is unchanged, as the quirk in its current form is overcosted at
2 points.

The emergent effect of skittish people diving into closets when caught
into explosions is definitely a feature, and not a bug.

cacogen commented:
Not sure how practical it would be but it's interesting

ATH1909 commented:
"tweak: The Skittish quirk will now cause you to automatically dive into
a locker/crate if you move into it while it is closed. Walk to avoid
this behaviour."
What if the locker/crate is open?

coiax commented:
> 
> 
> "tweak: The Skittish quirk will now cause you to automatically dive into
> a locker/crate if you move into it while it is closed. Walk to avoid
> this behaviour."
> What if the locker/crate is open?

Good point. It works if you "Bump" into the crate/locker. So it works if you run into a crate, regardless of it being open or closed, but only worked on closed lockers. I will update the CL.

Fikou commented on code/game/objects/structures/crates_lockers/closets.dm:
```diff
@@ -105,7 +105,7 @@
 		var/mob/living/L = user
 		if(HAS_TRAIT(L, TRAIT_SKITTISH))
-			. += "<span class='notice'>Ctrl-Shift-click [src] to jump inside.</span>"
+			. += "<span class='notice'>If you bump into [src] while running, you will jump inside.</span>"
```
i think simply "it" instead of [src] would sound better here?

Jared-Fogle replied:
`[src.p_them()]`

Pugie commented:
Why, it worked fine before?
At least until rohesie broke it with knockdown timer.

coiax committed:
Reworks skittish into a element

Skittish is now an element attached to mobs, when the TRAIT_SKITTISH is
added, rather than code that runs on every single Bump of a closet.

Some crates that don't function like normal crates, like the "loot
mastermind" crate, or the wooden crate than can only be deconstructed,
are not divable into.

coiax committed:
Calm linter gods

Rohesie commented on code/game/objects/structures/crates_lockers/closets.dm:
```diff
@@ -37,6 +37,7 @@
 	var/delivery_icon = "deliverycloset" //which icon to use when packagewrapped. null to be unwrappable.
 	var/anchorable = TRUE
 	var/icon_welded = "welded"
+	var/divable = TRUE /// whether a skittish person can dive inside this closet. disable if opening the closet causes "bad things" to happen.
```
```suggestion
	/// Whether a skittish person can dive inside this closet. disable if opening the closet causes "bad things" to happen.
	var/divable = TRUE
```

cacogen replied:
Also capitalise disable

Rohesie commented:
See if you can address the linter's issue.

coiax committed:
Update variable autodoc string

coiax committed:
Merge branch 'master' into dives-into-the-coffin

coiax committed:
Give silent argument on child proc definition

coiax committed:
Change Detach logic order

When subclassing destruction/detatching logic, should do our own logic
before calling super().

coiax committed:
Merge branch 'master' into dives-into-the-coffin

Rohesie commented on code/datums/elements/skittish.dm:
```diff
@@ -0,0 +1,61 @@
+
+	RegisterSignal(target, COMSIG_MOVABLE_BUMP, .proc/Bump)
+
+/datum/element/skittish/Detach(datum/target)
+	. = ..()
+	UnregisterSignal(target, COMSIG_MOVABLE_BUMP)
```
```suggestion
/datum/element/skittish/Detach(datum/target)
	UnregisterSignal(target, COMSIG_MOVABLE_BUMP)
	return ..()
```

coiax commented:
Looks like my VM's clock is wrong, so it appears I did these commits yesterday, but I did them just now.

Jared-Fogle commented on code/datums/traits/good.dm:
```diff
@@ -190,6 +190,7 @@
 	name = "Skittish"
 	desc = "You're easy to startle, and hide frequently. Run into a closed locker to jump into it, as long as you have access. You can walk to avoid this."
 	value = 2
+	value = 8
```
two values 😊

coiax committed:
Removes old skittish value

Left it in from the merge, whoops.

coiax committed:
Merge branch 'master' into dives-into-the-coffin
